python/tests/test_conversions.py

```diff
@@ -0,0 +1,181 @@
+# specific language governing permissions and limitations
+# under the License.
+
+import uuid
+from decimal import Decimal
+
+import pytest
+
+from iceberg import conversions
+from iceberg.types import (
+    BinaryType,
+    BooleanType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IntegerType,
+    LongType,
+    StringType,
+    TimestampType,
+    TimestamptzType,
+    TimeType,
+    UUIDType,
+)
+
+
+@pytest.mark.parametrize(
+    "type_, partition_value_as_str, expected_result",
+    [
+        (BooleanType(), "true", True),
+        (BooleanType(), "false", False),
+        (BooleanType(), "TRUE", True),
+        (BooleanType(), "FALSE", False),
+        (BooleanType(), None, False),
+        (IntegerType(), "1", 1),
+        (IntegerType(), "9999", 9999),
+        (LongType(), "123456789", 123456789),
+        (FloatType(), "1.1", 1.1),
+        (DoubleType(), "99999.9", 99999.9),
+        (StringType(), "foo", "foo"),
+        (UUIDType(), "f79c3e09-677c-4bbd-a479-3f349cb785e7", uuid.UUID("f79c3e09-677c-4bbd-a479-3f349cb785e7")),
+        (FixedType(3), "foo", bytearray(b"foo")),
+        (BinaryType(), "foo", b"foo"),
+        (None, "__HIVE_DEFAULT_PARTITION__", None),
+        (None, None, None),
+    ],
+)
+def test_from_partition_value_to_py(type_, partition_value_as_str, expected_result):
+    """Test converting a partition value to a python built-in"""
+    assert conversions.from_partition_value_to_py(type_, partition_value_as_str) == expected_result
+
+
+@pytest.mark.parametrize(
+    "type_, partition_value_as_str, expected_result",
+    [
+        (DecimalType(5, 2), "123.45", Decimal(123.45)),
+    ],
+)
+def test_from_partition_value_to_py_approximated(type_, partition_value_as_str, expected_result):
+    """Test approximate equality when converting a partition value to a python built-in"""
+    assert conversions.from_partition_value_to_py(type_, partition_value_as_str) == pytest.approx(expected_result)
+
+
+@pytest.mark.parametrize(
+    "type_, b, result",
+    [
+        (BooleanType(), b"\x00", False),
+        (BooleanType(), b"\x01", True),
+        (IntegerType(), b"\xd2\x04\x00\x00", 1234),
+        (LongType(), b"\xd2\x04\x00\x00\x00\x00\x00\x00", 1234),
+        (DateType(), b"\xd2\x04\x00\x00", 1234),
+        (TimeType(), b"\x00\xe8vH\x17\x00\x00\x00", 100000000000),
+        (TimestamptzType(), b"\x00\xe8vH\x17\x00\x00\x00", 100000000000),
+        (TimestampType(), b"\x00\xe8vH\x17\x00\x00\x00", 100000000000),
+        (StringType(), b"foo", "foo"),
+        (UUIDType(), b"\xf7\x9c>\tg|K\xbd\xa4y?4\x9c\xb7\x85\xe7", uuid.UUID("f79c3e09-677c-4bbd-a479-3f349cb785e7")),
+        (FixedType(3), b"foo", b"foo"),
+        (BinaryType(), b"foo", b"foo"),
+        (DecimalType(5, 2), b"\x30\x39", Decimal(123.45).quantize(Decimal(".01"))),
+        (DecimalType(5, 4), b"\x00\x12\xd6\x87", Decimal(123.4567).quantize(Decimal(".0001"))),
```

Comment:
similar comment here, I think it might be clearer to use string constructor on Decimal.

Reply:
Converted this everywhere in the tests so quantize isn't used anymore.